# HG changeset patch
# User Rik <rik@octave.org>
# Date 1399571961 25200
#      Thu May 08 10:59:21 2014 -0700
# Branch stable
# Node ID e5ab507687ee16e5479230f34d38e5e6d079bb04
# Parent  8b90d4be063bffba0c97edf10b10ac48f4003678
Correctly follow legend interpreter property for gnuplot (bug #40064).

* __go_draw_axes__.m: Use the 'interpreter' property of legend object
to set enhanced/noenhanced option to 'set key' gnuplot command.

diff --git a/scripts/plot/util/private/__go_draw_axes__.m b/scripts/plot/util/private/__go_draw_axes__.m
--- a/scripts/plot/util/private/__go_draw_axes__.m
+++ b/scripts/plot/util/private/__go_draw_axes__.m
@@ -1599,18 +1599,19 @@ function __go_draw_axes__ (h, plot_strea
     if (rows (textcolors) > 1)
       ## Gnuplot is unable to assign arbitrary colors to each text entry
       ## for the key/legend.  But, the text color can be set to match the
       ## color of the plot object.
       colorspec = "textcolor variable";
     else
       colorspec = get_text_colorspec (textcolors, mono);
     endif
-    fprintf (plot_stream, "set key %s %s;\nset key %s %s %s %s %s;\n",
-             inout, pos, box, reverse, horzvert, fontspec, colorspec);
+    fprintf (plot_stream, "set key %s %s;\nset key %s %s %s %s %s %s;\n",
+             inout, pos, box, reverse, horzvert, fontspec, colorspec,
+             __do_enhanced_option__ (enhanced, hlgnd));
   else
     fputs (plot_stream, "unset key;\n");
   endif
   fputs (plot_stream, "set style data lines;\n");
 
   cmap = [cmap; addedcmap];
   cmap_sz = cmap_sz + rows (addedcmap);
   if (length (cmap) > 0)
